### ISSUE TITLE
Descriptor loss parameters optimization 

### ISSUE THREAD

**Issue by idanshef** [rpautrat/SuperPoint#283]
Hi, your work on superpoint really helped me! Thank you for this great work 🙏

I saw that you changed lambda_loss and lambda_d with different parameters than defined in the paper.
How did you find the parameters which gave you the best results? A lot of experiments or there is another way?
And one more related question, I saw that you changed the lambdas, after you added L2 normalization to the descriptors product. Was there any transformation you did to those parameters or try and fail until you reached to optimal values?

Thanks in advance!!

**Comment by rpautrat** [comment]
Hi,

Unfortunately, there was not easier choice than just experimenting until I could find the best parameters. I cannot guarantee that the current values are optimal, but they should work fairly well at least. Feel free to keep tuning them if you want to improve the results further.

**Comment by Vincentqyw** [comment]
I have one similar question in tuning parameters, especially $\lambda$ and $\lambda_d$.  
Is there any critiria in tunning these two parameters when the trainning set is hard?

**Comment by rpautrat** [comment]
I don't see any reason for changing $\lambda$ and $\lambda_d$ when the training set is harder. The only reasons to change them would be the following I think:
- $\lambda$: if you care more for the quality of the keypoints or for the descriptors, you can change the balancing factor.
- $\lambda_d$: if you expect to have many keypoints in your training set, or on the contrary very few keypoints, you might want to tune $\lambda_d$ a bit.

**Comment by Vincentqyw** [comment]
Got it! Thanks for your advice.
